### PR TITLE
Support mu and sd parameterization for inverse gamma.

### PR DESCRIPTION
**Pull request by rpgoldman**
Since the Gamma supports `mu` and `sd` as an alternative parameterization, I added support for this to Inverse Gamma.
Hoping this is useful.

### CODE REVIEW

**Comment by twiecki** [comment]
This is great, but there are some PEP8 formatting issues.

**Comment by ColCarroll** [comment]
That's odd - the test fails only on Python 2.7, but passes on python 3. Any guesses? 

```
self = <pymc3.tests.test_distributions.TestMatchesScipy object at 0x7f90a205bbd0>
    def test_inverse_gamma(self):
        self.pymc3_matches_scipy(
            InverseGamma, Rplus, {'alpha': Rplus, 'beta': Rplus},
            lambda value, alpha, beta: sp.invgamma.logpdf(value, alpha, scale=beta))
        self.pymc3_matches_scipy(
            InverseGamma, Rplus, {'mu': Rplus, 'sd': Rplus},
>           lambda value, mu, sd: sp.invgamma.logpdf(value, InverseGamma._get_alpha_beta(None, None, mu, sd)[0],
                                                     scale=InverseGamma._get_alpha_beta(None, None, mu, sd)[1]))
pymc3/tests/test_distributions.py:706: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymc3/tests/test_distributions.py:438: in pymc3_matches_scipy
    self.check_logp(model, value, domain, paramdomains, logp, decimal=decimal)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <pymc3.tests.test_distributions.TestMatchesScipy object at 0x7f90a205bbd0>
model = <pymc3.model.Model object at 0x7f909465c610>, value = value
domain = <pymc3.tests.test_distributions.Domain object at 0x7f90b6404510>
paramdomains = {'mu': <pymc3.tests.test_distributions.Domain object at 0x7f90b6404510>, 'sd': <pymc3.tests.test_distributions.Domain object at 0x7f90b6404510>}
logp_reference = <function logp at 0x7f90a244eed8>, decimal = 3
    def check_logp(self, model, value, domain, paramdomains, logp_reference, decimal=None):
        domains = paramdomains.copy()
        domains['value'] = domain
        logp = model.fastlogp
        for pt in product(domains, n_samples=100):
            pt = Point(pt, model=model)
            if decimal is None:
                decimal = select_by_precision(float64=6, float32=3)
>           assert_almost_equal(logp(pt), logp_reference(pt), decimal=decimal, err_msg=str(pt))
E           AssertionError: 
E           Arrays are not almost equal to 3 decimals
E           {'mu': array(100., dtype=float32), 'value': array(0.9, dtype=float32), 'sd': array(1.5, dtype=float32)}
E           (mismatch 100.0%)
E            x: array(-468544.34, dtype=float32)
E            y: array(-468544.333)
pymc3/tests/test_distributions.py:448: AssertionError
```

**Comment by rpgoldman** [comment]
@twiecki 
> This is great, but there are some PEP8 formatting issues.

Thanks, but I just ran `autopep8` on the file, and AFAICT the PEP8 issues (mostly indentation) are not in parts of the file that I modified.  The only thing I can see is an extra newline.

`autopep8` doesn't like the way the latex representation methods are indented everywhere in this file, and there are a couple of places where it thinks lines are too long, but I don't see anything in my code.

Should I try to fix those (that seems like work for a different pull request), or is there something in my code that I'm missing?

**Comment by rpgoldman** [comment]
@ColCarroll 
> That's odd - the test fails only on Python 2.7, but passes on python 3. Any guesses?

I'm afraid I have no idea.  Is there any chance that python 2.7's floating point is less accurate than Python 3's, and that adding my additional floating point operation (to translate `mu` and `sd` to `alpha` and `beta`) causes a floating point mismatch?

I just ran the tests on my new MacBook Pro with Python 2.7 and `test_inverse_gamma` passed for me.

I note that although the error message says "accurate to 3 decimals," in this context we are demanding accuracy to 9 significant digits, which seems pretty demanding.  

This suggests that perhaps the values we are using as `mu` and `sd` (which are probably the same as the ones used for `alpha` and `beta`) aren't as appropriate for those parameters as for `alpha` and `beta`.

**Comment by rpgoldman** [comment]
@ColCarroll Looks like the new version works on Python 2.7 with 64-bit floats, and Python 3 with 32- and 64-bit, which suggests that this is some problem with Python 2.7 rather than PyMC3.

Can someone point me to an explanation for why, when `pymc3_matches_scipy` generates the test dictionaries with entries for `value, alpha, beta` we don't get value errors for inconsistent combinations?  When I try to iterate over `Rplus` to replicate what happens there (`Rplus` contains zero, but `mu=sd=0` is not a consistent parametrization for the Inverse Gamma. 

Is something in there trapping that?  Sorry: I'm afraid I'm not grokking how this test works.

**Comment by ColCarroll** [comment]
That is super strange that it is only python 2.7 float32 that fails. 

The test code is definitely a little older and hard to follow (rewriting it to use pytest fixtures would be nice, but also likely pretty tricky)

Let me see if I can get this to run on my machine before I lead you astray.

**Comment by ColCarroll** [comment]
good news/bad news: I can recreate the error on python 2.7 and python 3.6 (note that the float32 tests are all on python2.7).  So this seems to be a numerical accuracy issue. I'll dig a little more to see if I can fix it:


```
THEANO_FLAGS='floatX=float32' python -m pytest -xv pymc3/tests/test_distributions.py::TestMatchesScipy::test_inverse_gamma
```

**Comment by rpgoldman** [comment]
@ColCarroll I can replicate this on my mac.  With that setting of `THEANO_FLAGS` I get failures on both 2.7 and 3.6.

Since the value is so close to matching, I'm inclined to suspect that it was doing *my* translation options *in combination with* the built-in operations for sampling inverse gamma that caused the mismatch.

Speaking of which, is there a floating point type mismatch here:
```
E            x: array(-468544.34, dtype=float32)
E            y: array(-468544.333)
```
I note that the `y` value has no qualifier, so perhaps it's a 64-bit (or 128? what do I know?) quantity, and that accounts for the difference in precision?

File: pymc3/tests/test_distributions.py [hunk 1]
```diff
@@ -701,6 +701,10 @@ def test_inverse_gamma(self):
         self.pymc3_matches_scipy(
             InverseGamma, Rplus, {'alpha': Rplus, 'beta': Rplus},
             lambda value, alpha, beta: sp.invgamma.logpdf(value, alpha, scale=beta))
+        self.pymc3_matches_scipy(
```
Comment (ColCarroll) [inline pymc3/tests/test_distributions.py:711]: a little more readable:

```python
        def test_fun(value, mu, sd):
            alpha, beta = InverseGamma._get_alpha_beta(None, None, mu, sd)
            return sp.invgamma.logpdf(value, alpha, scale=beta)
        self.pymc3_matches_scipy(
            InverseGamma, Rplus, {'mu': Rplus, 'sd': Rplus}, test_fun)
```

File: pymc3/tests/test_distributions.py [hunk 2]
```diff
@@ -701,6 +701,10 @@ def test_inverse_gamma(self):
         self.pymc3_matches_scipy(
```
Comment (ColCarroll) [inline pymc3/tests/test_distributions.py]: I would decorate this with
```python
    @pytest.mark.xfail(condition=(theano.config.floatX == "float32"), reason="Fails on float32 due to scaling issues")
```

**Review by ColCarroll — COMMENTED** [review]
spent a bunch of time looking at this (and found an unrelated bug), but I think it is best to just punt on the float32 tests here.

**Comment by ColCarroll** [comment]
(yes, `scipy` is doing the calculation with full precision, and `theano` is using float32, and `mu=100`, `sd=0.9` makes `beta = 1,234,667.9`, which uses up a lot of the precision)

**Comment by rpgoldman** [comment]
OK, I made the recommended changes.  If this passes, I'd suggest squash merging it -- the history doesn't need all my missteps!
Still waiting to hear back from @twiecki -- if there are PEP8 fixes needed, I'm happy to make them.  But I don't know what they are.

**Comment by ColCarroll** [comment]
Thanks for the quick change! Will merge when it passes (pep8 looks good to me). We switched to squash and merge by default to make everything a little more manageable :)

**Comment by ColCarroll** [comment]
Thanks @rpgoldman!

**Comment by rpgoldman** [comment]
Thank you for all the help and advice!  Fun having a little bit of my code in PyMC3
